### PR TITLE
Add multiple power rings in pdngen for OpenFASOC's ldo support

### PR DESCRIPTION
**Pull request by minghungumich**
Create multiple power ring test case for regression.
Support multiple and parameterized power ring addition.
Signed-off-by: minghungumich <minghung@umich.edu>

### CODE REVIEW

**Comment by msaligane** [comment]
@Colin-Holehouse and @tspyrou This PR supersedes #1118 which has been sitting there too long. Could you please review/merge. We need this feature for the LDO in openfasoc.

**Comment by msaligane** [comment]
FYI. These tests are failing already before the PR:
```
[2021-11-30T02:54:08.785Z] tinyRocket *FAIL*

[2021-11-30T02:54:35.640Z] soc_bsg_black_parrot_nangate45 *FAIL*

[2021-11-30T02:55:07.525Z] soc_bsg_black_parrot_nangate45.pad_offset *FAIL*

[2021-11-30T02:55:07.525Z] extend_to_core_ring pass

[2021-11-30T02:55:12.878Z] asap7_vias *FAIL*

[2021-11-30T02:55:16.179Z] asap7_vias.noviarules *FAIL*
```

@minghungumich has sent an email discussing this.

**Comment by minghungumich** [comment]
> FYI. These tests are failing already before the PR:
> 
> ```
> [2021-11-30T02:54:08.785Z] tinyRocket *FAIL*
> 
> [2021-11-30T02:54:35.640Z] soc_bsg_black_parrot_nangate45 *FAIL*
> 
> [2021-11-30T02:55:07.525Z] soc_bsg_black_parrot_nangate45.pad_offset *FAIL*
> 
> [2021-11-30T02:55:07.525Z] extend_to_core_ring pass
> 
> [2021-11-30T02:55:12.878Z] asap7_vias *FAIL*
> 
> [2021-11-30T02:55:16.179Z] asap7_vias.noviarules *FAIL*
> ```
> 
> @minghungumich has sent an email - a couple weeks ago - discussing this.

I can easily fix these FAIL test cases by replacing the golden def file. Not sure if it is worth to fix these test cases.

**Comment by msaligane** [comment]
> I can easily fix these FAIL test cases by replacing the golden def file. Not sure if it is worth to fix these test cases.

Great! thank @minghungumich 
Lets see what @Colin-Holehouse thinks is best. Maybe there is something going on with asap7

**Comment by maliberty** [comment]
All tests are currently passing so this is related to your changes.

**Comment by msaligane** [comment]
@maliberty See Ming-Hung's commit `be49480`
That's why it is passing. Maybe you could review this PR? if it passes all the checks then maybe let's merge it?

File: src/pdn/test/regression [hunk 1]
```diff
@@ -26,6 +26,7 @@ set openroad_dir [file dirname [file dirname [file dirname $test_dir]]]
 
 source [file join $openroad_dir "test" "regression.tcl"]
 source [file join $test_dir "regression_tests.tcl"]
+source [file join $test_dir "regression_test.noapi.tcl"]
```
Comment (minghungumich) [inline src/pdn/test/regression:29]: @maliberty This is noapi regression test I mentioned. Previously it only checks for api test so not found the mismatch. But I think we should run both for regression test.

File: src/pdn/src/PdnGen.tcl [hunk 2]
```diff
@@ -179,13 +179,14 @@ proc define_pdn_grid {args} {
 sta::define_cmd_args "set_voltage_domain" {-name domain_name \
                                            [-region region_name] \
                                            -power power_net_name \
+                                           [-secondary_power secondary_power_net_name] \
```
Comment (arlpetergadfort) [inline src/pdn/src/PdnGen.tcl:182]: make sure to update the documentation to include this option

File: src/pdn/test/regression (hunk 1 above)
Comment (maliberty) [reply src/pdn/test/regression:29]: I think it is good to run the noapi test too.  However I don't understand why you are removing tests from noapi.tcl.   Is there a reason to not run them?

Comment (minghungumich) [reply src/pdn/test/regression:29]: I remove for now to pass the PR check. I will add them back after I fix them.

File: src/pdn/src/PdnGen.tcl (hunk 2 above)
Comment (minghungumich) [reply src/pdn/src/PdnGen.tcl:182]: Thanks. This is done.

File: src/pdn/test/regression (hunk 1 above)
Comment (maliberty) [reply src/pdn/test/regression:29]: Ok I'll approve and open an issue to track re-enabling them.

Comment (maliberty) [reply src/pdn/test/regression:29]: https://github.com/The-OpenROAD-Project/OpenROAD/issues/1373

**Review by maliberty — APPROVED** [review]

**Pull request merged**